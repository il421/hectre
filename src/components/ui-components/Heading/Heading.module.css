@value variables: "../../../common/variables.css";
@value font-size-xsmall, font-size-xxsmall, font-size-xxxsmall, xs-size from variables;

.common {
    margin: 0;
    padding: xs-size;

    text-transform: uppercase;
    font-weight: 500;
}

.h1 {
    composes: common;
    font-size: font-size-xsmall;
}

.h2 {
    composes: common;
    font-size: font-size-xxsmall;
}

.h3 {
    composes: common;
    font-size: font-size-xxxsmall;
}