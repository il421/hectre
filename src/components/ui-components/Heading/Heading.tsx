import { FunctionComponent } from "react";

import styles from "./Heading.module.css";

type Level = "main" | "second" | "third";

interface HeadingProps {
  level: Level;
}

export const Heading: FunctionComponent<HeadingProps> = ({
  level,
  children
}) => {
  switch (level) {
    case "main":
      return <h1 className={styles.h1}>{children}</h1>;
    case "second":
      return <h2 className={styles.h2}>{children}</h2>;
    default:
      return <h3 className={styles.h3}>{children}</h3>;
  }
};
