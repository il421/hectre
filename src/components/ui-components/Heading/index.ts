export * from "./Heading";
export * from "./Heading.module.css";
