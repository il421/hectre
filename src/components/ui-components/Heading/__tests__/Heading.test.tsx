import { shallow, ShallowWrapper } from "enzyme";

import { Heading } from "../Heading";

test("should match previous snapshot", () => {
  const wrapper: ShallowWrapper<typeof Heading> = shallow<typeof Heading>(
    <Heading level="second" />
  );
  expect(wrapper.debug()).toMatchSnapshot();
});
