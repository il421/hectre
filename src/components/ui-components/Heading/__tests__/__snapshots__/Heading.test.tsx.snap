// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `"<h2 className=\\"h2\\" />"`;
